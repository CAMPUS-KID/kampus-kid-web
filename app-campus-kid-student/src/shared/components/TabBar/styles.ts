import { css } from '@emotion/native';
import { Theme } from '@theme';

export const styleSheet = ({ colors }: Theme) => ({
  container: css`
    flex-direction: row;
    border-top-width: 1px;
    border-top-color: ${colors.shadowGray};
    padding: 0px 20px;
  `
});
