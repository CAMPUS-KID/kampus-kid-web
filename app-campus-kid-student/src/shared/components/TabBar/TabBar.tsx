import React from 'react';
import { SafeAreaView } from 'react-native-safe-area-context';
import { useTranslation } from 'react-i18next';

import { useBuildStyles, useTesting } from '@shared_hooks';
import { Routes, Components } from '@shared_constants';
import { styleSheet } from './styles';
import { TabMenuItem } from '../TabMenuItem';
import { BottomTabBarProps } from '@react-navigation/bottom-tabs';
import { IconEnum } from '@shared_enums';

const TabBar = ({ state: { index } }: BottomTabBarProps) => {
  const styles = useBuildStyles(styleSheet);
  const { t } = useTranslation();
  const { getTestingProps } = useTesting(Components.TAB_BAR);

  return (
    <SafeAreaView style={styles.container} edges={['bottom']}>
      <TabMenuItem
        {...getTestingProps(`tab-item-movies`)}
        route={Routes.MOVIES.MOVIE_LIST}
        icon={IconEnum.MOVIES}
        tabName={t('tabs:movies')}
        isActive={index === 0}
      />
      <TabMenuItem
        {...getTestingProps(`tab-item-series`)}
        route={Routes.SERIES.SERIE_LIST}
        icon={IconEnum.SERIES}
        tabName={t('tabs:series')}
        isActive={index === 1}
      />
    </SafeAreaView>
  );
};
export default TabBar;
