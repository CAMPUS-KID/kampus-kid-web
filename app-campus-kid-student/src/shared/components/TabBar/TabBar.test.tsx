import React from 'react';
import TabBar from './TabBar';
import { render } from 'react-native-testing-library';
import { useStackNavigation } from '@shared_hooks';

// eslint-disable-next-line react-hooks/rules-of-hooks
const navigation = useStackNavigation();

describe('TabBar', () => {
  let props;

  beforeEach(() => {
    props = {
      state: {
        index: 0
      },
      navigation
    };
  });

  it('should render', () => {
    const { toJSON } = render(<TabBar {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
