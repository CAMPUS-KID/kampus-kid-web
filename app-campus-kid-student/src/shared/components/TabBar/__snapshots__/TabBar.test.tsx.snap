// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar should render 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
    ]
  }
  style={
    Object {
      "borderTopColor": "#EEEEEE",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "paddingBottom": 0,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 0,
    }
  }
>
  <View
    accessibilityLabel="tab item"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 10,
      }
    }
    testID="tab-bar-item-tab-item"
  >
    <SvgImage
      color="#F00303"
      fill="#F00303"
      height={30}
      width={30}
    />
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Inter",
            "fontSize": 10,
          },
          Object {
            "color": "#F00303",
          },
        ]
      }
    >
      translate(tabs:movies)
    </Text>
  </View>
  <View
    accessibilityLabel="tab item"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 10,
      }
    }
    testID="tab-bar-item-tab-item"
  >
    <SvgImage
      color="#8A8A8A"
      fill="#8A8A8A"
      height={30}
      width={30}
    />
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Inter",
            "fontSize": 10,
          },
          Object {
            "color": "#8A8A8A",
          },
        ]
      }
    >
      translate(tabs:series)
    </Text>
  </View>
</RNCSafeAreaView>
`;
